src/Common/src/TypeSystem/Common/DefType.FieldLayout.cs

```diff
@@ -275,6 +285,22 @@ public DefType HfaElementType
             }
         }
 
+        /// <summary>
+        /// Gets a value indicating whether this is a byref-like type
+        /// (a <code>TypedReference</code>, <code>Span&lt;T&gt;</code>, etc.).
+        /// </summary>
+        public bool IsByRefLike
+        {
+            get
+            {
+                if (!_fieldLayoutFlags.HasFlags(FieldLayoutFlags.ComputedIsByRefLike))
+                {
+                    ComputeIsByRefLike();
+                }
+                return _fieldLayoutFlags.HasFlags(FieldLayoutFlags.IsByRefLike);
```

Comment:
BTW: I expect that the ref-like types will be eventually identified by attribute, and so ComputeIsByRefLike will be just a check of the attribute; not a property of the field layout.